webapp/src/actions/index.ts

```diff
@@ -217,6 +217,48 @@ export const createChannelSubscription = (subscription: ChannelSubscription) =>
     };
 };
 
+export const createSubscriptionTemplate = (subscription: ChannelSubscription) => {
+    return async (dispatch, getState) => {
+        const baseUrl = getPluginServerRoute(getState());
+        try {
+            const data = await doFetch(`${baseUrl}/api/v2/subscriptionTemplates?channel_id=${subscription.channel_id}`, {
+                method: 'post',
+                body: JSON.stringify(subscription),
+            });
+
+            dispatch({
+                type: ActionTypes.CREATED_SUBSCRIPTION_TEMPLATE,
+                data,
+            });
+
+            return {data};
+        } catch (error) {
+            return {error};
+        }
+    };
```

Comment:
you should create a common async function/wrapper for this `doFetch` and call here/wherever required with args like `url, method, body, actionType`

Reply:
Every Api call in this plugin is done like that. I have added 2-3 Api call and creating a wrapper for only those API as we cannot change the previously written code and it will also affect the consistency of the code.

Reply:
Okay considering that same is done across the project we can leave it